src/lexer/lexer.c

```diff
@@ -1,10 +1,32 @@
 #include "minishell.h"
 
+static char	*new_quotes_remove(char *str, int quote_i)
+{
+	char	**split_words;
+	char	*new;
+
+	split_words = new_str_split(str, str[quote_i]);
+	new = new_str_join(split_words, '\0');
+	del_arr(split_words);
+	return (new);
+}
+
+static char	*new_expand_parameter(char *str)
+{
+	char	*new;
+
+	new = new_str("\"");
+	ft_str_append(&new, str);
+	ft_str_append(&new, "\"");
+	return (new);
```

Comment:
단일 문자열을 합쳐서 반환한다면 `new_str_join()` 함수를 써도 좋을 것 같습니다

```suggestion
	return (new_str_join((char *[]){"\"", str, "\"", NULL}, '\0'));
```